src/common/load-plugins.js

```diff
@@ -25,11 +26,48 @@ function load(plugins, pluginSearchDirs) {
   if (!pluginSearchDirs) {
     pluginSearchDirs = [];
   }
-  // unless pluginSearchDirs are provided, auto-load plugins from node_modules that are parent to Prettier
+  // unless pluginSearchDirs are provided, auto-load plugins from
+  // the first node_modules folder that is parent to the cwd
+
+  const findPluginDirsInPnpmVirtualStore = (searchDirectory) => {
+    const pluginPackageJsonPaths = fastGlob.sync(
+      ["prettier-plugin-*", "@*/node_modules"],
+      {
+        cwd: searchDirectory,
+        onlyDirectories: true,
+      }
+    );
+    return pluginPackageJsonPaths.map((match) =>
+      path.resolve(searchDirectory, match)
+    );
+  };
+
   if (pluginSearchDirs.length === 0) {
-    const autoLoadDir = thirdParty.findParentDir(__dirname, "node_modules");
-    if (autoLoadDir) {
-      pluginSearchDirs = [autoLoadDir];
+    // This is likely the project root. In projects using pnpm, this directory is named "prettier@<version>".
+    let autoLoadDirectory = thirdParty.findParentDir(__dirname, "node_modules");
+    autoLoadDirectory = autoLoadDirectory
+      ? path.resolve(autoLoadDirectory)
+      : ""; // Make sure autoLoadDir isn't null.
+
+    const isProjectRoot = findProjectRoot(__dirname) === autoLoadDirectory; // Make sure we don't go outside projectRoot.
+
+    // Get the parent directory of autoLoadDir.
+    // It will be the pnpm virtual store if we're not at the project root and if "prettier@" is found.
+    const pnpmVirtualStore =
+      !isProjectRoot &&
+      autoLoadDirectory.includes("prettier@") &&
```

Comment:
`path.dirname(...).startsWith("prettier@")`

Reply:
@fisker `pnpmVirtualStore` gets set to the path returned by `path.dirname()` (assuming the first two checks pass). It's not supposed to be `true`. I might be misunderstanding your comment though.

I'll change `autoLoadDirectory.includes("prettier@")` to `autoLoadDirectory.startWith("prettier@")`.

Reply:
Isn't it a absolute path? Are we testing the dir name is `pretty@2.5.1`?

Reply:
When using pnpm, the parent directory of `autoLoadDirectory` is the pnpm virtual store. We check to see if `autoLoadDirectory` starts with `prettier@` so we know that pnpm is being used. Then we grab the parent directory with `path.dirname`.

Reply:
I understand that. `autoLoadDirectory` is a absolute path, `.startsWith()` will never be true, you need get the dirname.

Reply:
We can know the version of Prettier by reading `package.json` so we can test the dirname is exactly `prettier@<version>`

Reply:
Ah, gotcha, sorry. Pushed a commit that should fix that. Would it not be better/less fragile to just check for `startsWith("prettier@")`. (For example, if pnpm changes how it appends the version number.)